### PR TITLE
fix recent emoji pane

### PR DESCRIPTION
**Pull request by mcginty**
1) Make recent list properly update and invalidate.
2) Show most-recently-used first.
3) As a bonus, make drafts and emoji get along.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/components/emoji/EmojiPageModel.java [hunk 1]
```diff
@@ -1,7 +1,12 @@
 package org.thoughtcrime.securesms.components.emoji;
 
-public interface EmojiPageModel {
-  int getIconRes();
-  int[] getCodePoints();
-  void onCodePointSelected(int codePoint);
+public abstract class EmojiPageModel {
+  public abstract int getIconRes();
+  public abstract int[] getCodePoints();
+  public void onCodePointSelected(int codePoint) { }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/emoji/EmojiPageModel.java]: does this need to be in the interface?

File: src/org/thoughtcrime/securesms/components/emoji/RecentEmojiPageModel.java [hunk 2]
```diff
@@ -50,10 +51,11 @@ public RecentEmojiPageModel(Context context) {
   }
 
   @Override public int[] getCodePoints() {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/emoji/RecentEmojiPageModel.java]: why are these primitive arrays and not lists?

**Comment by moxie0** [comment]
Just looked at EmojiProvider, what is all that executor service invalidate drawable stuff?  I think EmojiProvider should be synchronous, and anything wanting asynchronous loads should be using a loader.

File: src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java [hunk 3]
```diff
@@ -25,6 +25,13 @@ public EmojiEditText(Context context, AttributeSet attrs,
     init();
   }
 
+  @Override public void setText(CharSequence text, BufferType type) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java]: What is the purpose of the insertEmoji method if you're overriding this?

Comment (mcginty) [inline src/org/thoughtcrime/securesms/components/emoji/EmojiEditText.java]: getText().replace() should be more efficient, as we don't need to run the regex replacement over the entire text when inserting.

**Pull request closed without merging**